src/App.tsx

```diff
@@ -1,61 +1,145 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+type Togo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+};
+
+type Data =
+  | React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLSelectElement>;
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [todos, setTodos] = useState<Togo[]>(todosFromServer);
+
+  const [userHasError, setUserHasError] = useState(false);
+  const [titleHasError, setTitleHasError] = useState(false);
+
+  const handleChange = (e: Data) => {
+    if (e.target.name === 'title') {
+      setTitle(e.target.value);
+      setTitleHasError(false);
+    } else {
+      setUserId(Number(e.target.value));
+      setUserHasError(false);
+    }
+  };
+
+  const checkTitle = () => {
+    return title.trim().length > 0;
+  };
+
+  function getNewId() {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  }
+
+  const addTodo = () => {
+    const user = usersFromServer.find(person => person.id === userId);
+
+    const id = user?.id || 0;
```

Comment:
The logic for setting `id` in `addTodo` might be incorrect. If no user is found, `id` will be set to 0, which might not be the intended behavior. Consider handling the case where no user is found more explicitly.